test/e2e/test_system.py

```diff
@@ -4,603 +4,579 @@
 
 from test.conftest import add_ids_to_properties
 from test.e2e.conftest import replace_unit_values_with_ids_in_properties
-from test.e2e.mock_schemas import (
-    CREATED_MODIFIED_VALUES_EXPECTED,
-    SYSTEM_POST_A,
-    SYSTEM_POST_A_EXPECTED,
-    SYSTEM_POST_B,
-    SYSTEM_POST_B_EXPECTED,
-    SYSTEM_POST_C,
-    USAGE_STATUS_POST_B,
-)
+from test.e2e.mock_schemas import USAGE_STATUS_POST_B
 from test.e2e.test_catalogue_item import CATALOGUE_CATEGORY_POST_A, CATALOGUE_ITEM_POST_A
 from test.e2e.test_item import ITEM_POST, MANUFACTURER_POST
 from test.e2e.test_unit import UNIT_POST_A, UNIT_POST_B
-from typing import Any, Optional
-from unittest.mock import ANY
+from test.mock_data import (
+    SYSTEM_GET_DATA_ALL_VALUES_NO_PARENT,
+    SYSTEM_GET_DATA_REQUIRED_VALUES_ONLY,
+    SYSTEM_POST_DATA_ALL_VALUES_NO_PARENT,
+    SYSTEM_POST_DATA_REQUIRED_VALUES_ONLY,
+)
+from typing import Optional
 
+import pytest
 from bson import ObjectId
+from fastapi import Response
+from fastapi.testclient import TestClient
 
 from inventory_management_system_api.core.consts import BREADCRUMBS_TRAIL_MAX_LENGTH
 
-SYSTEM_POST_REQUIRED_ONLY = {
-    "name": "System Test",
-    "importance": "low",
-}
 
-SYSTEM_POST_REQUIRED_ONLY_EXPECTED = {
-    **SYSTEM_POST_REQUIRED_ONLY,
-    **CREATED_MODIFIED_VALUES_EXPECTED,
-    "id": ANY,
-    "parent_id": None,
-    "description": None,
-    "location": None,
-    "owner": None,
-    "code": "system-test",
-}
+class CreateDSL:
+    """Base class for create tests"""
+
+    test_client: TestClient
+
+    _post_response: list[Response]
+
+    @pytest.fixture(autouse=True)
+    def setup(self, test_client):
+        """Setup fixtures"""
+
+        self.test_client = test_client
+        self._post_response = []
+
+    def post_system(self, system_post_data: dict) -> Optional[str]:
+        """Posts a System with the given data, returns the id of the created system if successful
+
+        :param system_post_data: Dictionary containing the system data that should be posted
+        :return: ID of the created system (or None if not successful)
+        """
+        self._post_response = self.test_client.post("/v1/systems", json=system_post_data)
```

Comment:
It's complaining about the type - `Expected type 'list[Response]', got 'Response' instead`.